src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -15,6 +15,35 @@
 
 static inline int MPIDI_OFI_progress_do_queue(int vci_idx);
 
+/* Acquire a sequence number to send, and record the next number */
+MPL_STATIC_INLINE_PREFIX uint16_t MPIDI_OFI_am_fetch_incr_send_seqno(MPIR_Comm * comm,
+                                                                     int dest_rank)
+{
+    fi_addr_t addr = MPIDI_OFI_comm_to_phys(comm, dest_rank);
+    uint64_t id = addr;
+    uint16_t seq, old_seq;
+    void *ret;
+    ret = MPIDIU_map_lookup(MPIDI_OFI_global.am_send_seq_tracker, id);
+    if (ret == MPIDIU_MAP_NOT_FOUND)
+        old_seq = 0;
+    else
+        old_seq = (uint16_t) (uintptr_t) ret;
```

Comment:
This type casting gives me a little bit of a headache, but I suppose it is convenient to re-use the map structure as-is. An anonymous union would go a long way to making this more readable, but I don't think we're quite ready to require that in MPICH. Perhaps we can think of a way to improve the map API to better support types in the future. Otherwise, this looks OK to me. I'm going to signoff and merge.